src/ArmTemplates/Common/API/Clients/Groups/GroupsClient.cs

```diff
@@ -17,15 +17,27 @@ namespace Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.API.Clien
     public class GroupsClient : ApiClientBase, IGroupsClient
     {
         const string GetAllGroupsLinkedToProductRequest = "{0}/subscriptions/{1}/resourceGroups/{2}/providers/Microsoft.ApiManagement/service/{3}/products/{4}/groups?api-version={5}";
+        const string GetAllRequest = "{0}/subscriptions/{1}/resourceGroups/{2}/providers/Microsoft.ApiManagement/service/{3}/groups?api-version={4}";
+
+        public async Task<List<GroupTemplateResource>> GetAllAsync(ExtractorParameters extractorParameters)
+        {
+            var (azToken, azSubId) = await this.Auth.GetAccessToken();
+
+            var requestUrl = string.Format(GetAllRequest,
+                this.BaseUrl, azSubId, extractorParameters.ResourceGroup, extractorParameters.SourceApimName, GlobalConstants.ApiVersion);
+
+            var response = await this.CallApiManagementAsync<GetGroupsResponse>(azToken, requestUrl);
+            return response.Groups;
+        }
```

Comment:
We can merge both methods so that they are both reduced to this:
```suggestion
        public async Task<List<GroupTemplateResource>> GetAllAsync(ExtractorParameters extractorParameters)
        {
            var requestUrl = string.Format(GetAllRequest,
                this.BaseUrl, azSubId, extractorParameters.ResourceGroup, extractorParameters.SourceApimName, GlobalConstants.ApiVersion);

            return await this.GetGroupsFromApiAsync(requestUrl);
        }
        public async Task<List<GroupTemplateResource>> GetGroupsFromApiAsync(string requestUrl)
        {
            var (azToken, azSubId) = await this.Auth.GetAccessToken();
            var requestUrl = string.Format(GetAllRequest,
                this.BaseUrl, azSubId, extractorParameters.ResourceGroup, extractorParameters.SourceApimName, GlobalConstants.ApiVersion);

            return await this.CallApiManagementAsync<GetGroupsResponse>(azToken, requestUrl).Groups;
        }
```

This is quick and dirty to just should the idea though

Reply:
Optionally, since every client calls `await this.Auth.GetAccessToken()`; just move that to `CallApiManagementAsync` which is maybe a better approach for the above suggestion

Reply:
As for me I would leave it because we will possible use a [Microsoft.Azure.Management.ApiManagement](https://github.com/Azure/azure-api-management-devops-resource-kit/issues/605) NuGet instead of making raw HttpClient calls. And I will change every explicit call just to a method call in a nuget. That is why it would be better to leave every method not touched and not dependent on any request url IMO.

But ye, if we are staying with HttpClient approach, I will change it in future because I also don't like duplicated code here